src/ngx_http_lua_ssl_certby.c

```diff
@@ -880,6 +880,72 @@ ngx_http_lua_ffi_ssl_server_name(ngx_http_request_t *r, char **name,
 #endif
 }
 
+int
+ngx_http_lua_ffi_ssl_raw_client_addr(ngx_http_request_t *r, char **addr,
+    size_t *addrlen, int *addrtype, char **err)
+{
+#if (NGX_HAVE_UNIX_DOMAIN)
+    struct sockaddr_un *saun;
+#endif
+    ngx_ssl_conn_t *ssl_conn;
+    ngx_connection_t *c;
+    struct sockaddr_in *sin;
+#if (NGX_HAVE_INET6)
+    struct sockaddr_in6 *sin6;
+#endif
+
+    if (r->connection == NULL || r->connection->ssl == NULL) {
+        *err = "bad request";
+        return NGX_ERROR;
+    }
+
+    ssl_conn = r->connection->ssl->connection;
+    if (ssl_conn == NULL) {
+        *err = "bad ssl conn";
+        return NGX_ERROR;
+    }
+
+    c = ngx_ssl_get_connection(ssl_conn);
+
+    switch (c->sockaddr->sa_family) {
+
+#if (NGX_HAVE_INET6)
+    case AF_INET6:
+        sin6 = (struct sockaddr_in6 *) c->sockaddr;
+        *addrlen = 16;
+        *addr = (char *) &sin6->sin6_addr.s6_addr;
+        *addrtype = NGX_HTTP_LUA_ADDR_TYPE_INET6;
+
+        break;
+#endif
+
+# if(NGX_HAVE_UNIX_DOMAIN)
+    case AF_UNIX:
+        saun = (struct sockaddr_un *)c->sockaddr;
+        /* on Linux sockaddr might not include sun_path at all */ 
+        if(c->socklen <= (socklen_t) offsetof(struct sockaddr_un, sun_path))
+        {
+            *addr = "";
+            *addrlen = 0;
+        } else {
```

Comment:
Style: need a blank line before this `} else {` line. Please see the rest of the code base (and also the official nginx core) for examples.